eventbus/kafka/eventbus.go

```diff
@@ -314,30 +319,37 @@ func (b *EventBus) handle(m eh.EventMatcher, h eh.EventHandler, r *kafka.Reader)
 			continue
 		}
 
-		if err := handler(b.cctx, msg); err != nil {
+		for {
 			select {
-			case b.errCh <- err:
+			case <-b.cctx.Done():
+				return
 			default:
-				log.Printf("eventhorizon: missed error in Kafka event bus: %s", err)
 			}
 
-			continue
-		}
+			if err := handler(b.cctx, msg); err != nil {
+				select {
+				case b.errCh <- err:
+				default:
+					log.Printf("eventhorizon: missed error in Kafka event bus: %s", err)
+				}
+
+				time.Sleep(time.Second)
```

Comment:
Is this supposed to be here?

Reply:
This is similar to approach in the GCP implementation.